substrate/bin/node-template/runtime/src/lib.rs

```diff
@@ -279,19 +279,42 @@ impl pallet_template::Config for Runtime {
 }
 
 // Create the runtime by composing the FRAME pallets that were previously configured.
-construct_runtime!(
-	pub struct Runtime {
+#[frame_support::construct_runtime_v2]
+mod runtime {
+	#[frame::runtime]
+	pub struct Runtime;
+
+	#[frame::pallets]
+	#[frame::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId
+	)]
```

Comment:
yeah this could literally just be `#[derive(..)]` with slightly more parsing effort